hedera-node/src/main/java/com/hedera/services/state/logic/RecordStreaming.java

```diff
@@ -112,11 +118,45 @@ public void streamSystemRecord(final RecordStreamObject rso) {
     }
 
     private void stream(final RecordStreamObject rso) {
+        if (blockManager.shouldLogEveryTransaction()) {
+            logTransaction(rso);
+        }
         while (!nonBlockingHandoff.offer(rso)) {
             // Cannot proceed until we have handed off the record.
         }
     }
 
+    private void logTransaction(final RecordStreamObject rso) {
+        final var consTimestamp = rso.getTimestamp().toString();
+        final var blockNumber = rso.getStreamAlignment();
+        final var txId = rso.getTransactionRecord().getTransactionID();
+        final var txIdString =
+                String.format(
```

Comment:
Since this is a debug tool, performance doesn't matter much; but note that `String.format()` is very slow compared to string concatenation.

Reply:
Sure, I can change it to a `StringBuilder` or concatenation. I usually go for `String.format()` because it provides the best readability of the 3, but here I guess the readability won't really be an issue.